dcpy/utils/s3.py

```diff
@@ -134,24 +133,26 @@ def copy_file(
     source_key: str,
     target_key: str,
     acl: str,
+    *,
     target_bucket: Optional[str] = None,
+    metadata: Optional[dict] = None,
 ):
+    """Copies a file from from one location in S3 to another"""
     if target_bucket is None:
         target_bucket = bucket
-    """Copies a file from from one location in S3 to another"""
-    with _progress() as progress:
-        client_ = client()
-        size = client_.head_object(Bucket=bucket, Key=source_key)["ContentLength"]
-        task = progress.add_task(
-            f"[green]Copying [bold]{Path(source_key).name}[/bold]", total=size
-        )
-        client_.copy(
-            CopySource={"Bucket": bucket, "Key": source_key},
-            Bucket=target_bucket,
-            Key=target_key,
-            ExtraArgs={"ACL": acl},
-            Callback=lambda bytes: progress.update(task, advance=bytes),
-        )
+    client_ = client()
+    all_metadata = client_.head_object(Bucket=bucket, Key=source_key)
+    meta = all_metadata.get("Metadata", {})
+    if metadata is not None:
+        meta.update(metadata)
+    client_.copy_object(
+        CopySource={"Bucket": bucket, "Key": source_key},
+        Bucket=target_bucket,
+        Key=target_key,
+        ACL=acl,
+        Metadata=meta,
+        MetadataDirective="REPLACE",
```

Comment:
Otherwise, supplied metadata is ignored, only metadata of copied object is used. Relevant in case of "publishing" old versions this week - I wanted to set the "date-created" to the current "last-modified" as to not lose that information